pkg/common/checkeractionwatchdog.go

```diff
@@ -92,6 +95,36 @@ func (w *checkerActionWatchdog) Start(ctx context.Context) {
 	}()
 }
 
+// Confirm gives the confirmation that condition checked by the watchdog has failed or not.
+// It returns the boolean.
+func (w *checkerActionWatchdog) Confirm(ctx context.Context, failureThreshold uint, backoff *backoff.ExponentialBackoff) bool {
+	w.logger.Debug("Starting watchdog confirm")
+	defer w.logger.Debug("Stopping watchdog confirm")
+
+	ctx, w.cancelFunc = context.WithCancel(ctx)
```

Comment:
Please don't overwrite the `cancelFunc` here because it rather belongs to the watchdog's context which is closed by the `Stop()` function. I think it's sufficient if you directly use the context `ctx` that is passed.

Reply:
yes, it’s just I forgot to remove